### PR TITLE
2109 auth island version

### LINKED ISSUE #2109: Island Version checking
# Refactor

## Component(s) to be refactored

- monkey_island.cc.resources.version_update
- monkey_island.cc.services.version_update

## Tasks
- [x] Add a new object called `Version` with attributes: (0d) @ilija-lazoroski 
  - [x] `version_number` - Passed in to `Version()` constructor
  - [x] `latest_version` - A property that is populated asynchronously upon construction
  - [x] `download_url`- A property that is populated asynchronously upon construction
- [x] Construct `Version()` at Island startup (0d) @ilija-lazoroski 
- [x] Rename VersionUpdate to Version (0d) @ilija-lazoroski 
  - [x] Rename `/api/version-update` to `/api/island/version`
  - [x] Rename `VersionUpdate` to `Version`
  - [x] Rename `resources/version_update.py` to `resources/version.py`
- [x] Use `Version` object in `/api/island/version` endpoint (0d) @ilija-lazoroski 
  - [x] Add `Version` object to DIContainer
  - [x] Rename `newer_version` to `latest_version`
  - [x] Rename `current_version` to `version_number`
  - [x] Modify the UI to compare `version_number` to `latest_version`
  - [x] Add `Version` to `resources/__init__.py` and shorten all imports
- [x] Remove `services/version_update.py` (0d) @ilija-lazoroski 
- [x] Add authentication to `/api/island/version`. There are comments stating that the UI needs to know the version number before authentication, but I don't know why that's the case. (0d)

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Closes #2109.

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?
* [ ] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/monkey_island/cc/ui/src/components/logo/VersionComponent.js [hunk 1]
```diff
@@ -14,7 +15,7 @@ class VersionComponent extends React.Component {
   }
 
   componentDidMount() {
-    fetch('/api/island/version') // This is not authenticated on purpose
+    this.authFetch('/api/island/version') // This is not authenticated on purpose
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/ui/src/components/logo/VersionComponent.js:18]:
```suggestion
    this.authFetch('/api/island/version')
```

File: monkey/monkey_island/cc/ui/src/components/logo/VersionComponent.js [hunk 2]
```diff
@@ -27,6 +28,7 @@ class VersionComponent extends React.Component {
 
   newerVersionAvailable() {
     const semverGt = require('semver/functions/gt');
+    console.log(this.state);
```
Comment (VakarisZ) [inline monkey/monkey_island/cc/ui/src/components/logo/VersionComponent.js:31]:
```suggestion
```

**Review by VakarisZ — CHANGES_REQUESTED** [review]
Small changes

**Review by mssalvatore — APPROVED** [review]
LGTM, as long as it's tested and @VakarisZ approves.

**Review by VakarisZ — APPROVED** [review]

**Pull request merged**